lib/core/presentation/pages/event/guest_event_detail_page/widgets/guest_event_location.dart

```diff
@@ -29,77 +29,96 @@ class GuestEventLocation extends StatelessWidget {
         authenticated: (authSession) => authSession.userId, orElse: () => '');
     final isAttending = (event.accepted ?? []).contains(userId);
 
-    return SizedBox(
-      width: double.infinity,
-      height: 144.w,
-      child: Row(
-        children: [
-          SizedBox(
-            width: 144.w,
-            child: ClipRRect(
-                borderRadius: BorderRadius.only(
-                  topLeft: Radius.circular(15.sp),
-                  bottomLeft: Radius.circular(15.sp),
+    return FutureBuilder(
+      future: isAttending
+          ? MapUtils.getLocationName(
+              lat: event.latitude ?? 0, lng: event.longitude ?? 0)
+          : Future.delayed(const Duration(milliseconds: 300), () => ''),
+      builder: (context, snapshot) => SizedBox(
```

Comment:
Why do we need an empty future method here?

Reply:
I dont want to call gg service if user not attend the event yet, so jsut a fake future , the value of emptry will not be used